src/pal/src/CMakeLists.txt

```diff
@@ -70,6 +63,16 @@ elseif(PAL_CMAKE_PLATFORM_ARCH_ARM64)
   add_definitions(-D_WIN64=1)
 endif()
 
+if(CMAKE_SYSTEM_NAME STREQUAL Linux AND NOT CLR_CMAKE_PLATFORM_ALPINE_LINUX)
+  # Currently the _xstate is not available on Alpine Linux
+  add_definitions(-DXSTATE_SUPPORTED)
+endif(CMAKE_SYSTEM_NAME STREQUAL Linux AND NOT CLR_CMAKE_PLATFORM_ALPINE_LINUX)
+
+if(CLR_CMAKE_PLATFORM_ALPINE_LINUX)
+  # Setting RLIMIT_NOFILE breaks debugging of coreclr on Alpine Linux for some reason
```

Comment:
@sergiy-k it makes sense. One interesting thing that I've just tried is that if I have a simple testing app that just does the getrlimit / setrlimit like we do here, it doesn't have this effect on GDB. So maybe it is because we are calling it from a shared library or something else.
Btw, On Ubuntu, the rlim_cur is 0x400 and rlim_max is 0x1000. So on Alpine, even the rlim_cur is 128 times higher than the rlim_max on Ubuntu. So we really don't need to fiddle with it on Alpine.